src/cli/cmd.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+var path = require('path');
+var async = require('async');
+var db = require('../database');
+var plugins = require('../plugins');
+
+function parseCommands(pluginList, callback) {
+	var commands = [];
+
+	pluginList.forEach(function (pluginData) {
+		var pluginCommands = pluginData.commands || [];
+
+		pluginCommands.forEach(function (cmdData) {
+			var pluginName = pluginData.id;
+			var libraryFile = cmdData.library ? cmdData.library : pluginData.library;
+			var cmdName = pluginName.replace('nodebb-plugin-', '') + ':' + cmdData.cmd;
+			var scriptFile = path.resolve(pluginData.path, libraryFile);
+			var action = require(scriptFile)[cmdData.method];
+
+			commands.push({
+				plugin: pluginName,
+				name: cmdName,
+				description: cmdData.description,
+				options: cmdData.options,
+				scriptFile: scriptFile,
+				action: action,
+			});
+		});
+	});
+	callback(null, commands);
+}
+
+
+function start(command, args, program) {
+	var subProgram = new program.Command('cmd');
+
+	subProgram.on('command:*', function () {
+		console.error('Invalid command: %s\nSee --help for a list of available commands.', program.args.join(' '));
+		process.exit(1);
+	});
+
+	// remove the cmd segment from argvs
+	var argv = process.argv.slice(0, 2).concat(command, args);
+	command = command.replace(/^nodebb-plugin/, '');
+
+	async.waterfall(
+		[
```

Comment:
Remove the new lines around the function arguments to save one layer of indent.

Reply:
Ok https://github.com/NodeBB/NodeBB/pull/6901/commits/f8306711c600f3dcea7011a45b6f9adbdc1fc927